File: TypeScript/ArrayProducts/arrayProducts.ts
function productExceptSelf(nums: number[]): number[] {
  let answer: number[] = [];

  for (let i = 0; i < nums.length; i++) {
      let newArr: number[] = nums.filter(n => n != nums[i]);
      let prod: number = 1;
      for (let j = 0; j < newArr.length; j++) {
          prod = prod * newArr[j]
      }
      answer.push(prod)
  }

  return answer;
};

File: Java/ArrayProducts/ArrayProducts.java
package Java.ArrayProducts;

import java.lang.reflect.Array;
import java.util.ArrayList;
import java.util.Arrays;

public class ArrayProducts {

  public int productExceptSelf(Integer[] nums) {
    // ArrayList<Integer> answer = new ArrayList<Integer>(Arrays.asList(nums));
    int answer = 5;
    for (int i = 0; i < nums.length; i++) {
      Integer[] newArr = (Integer[]) Arrays.stream(nums).filter(x -> x != Array.get(nums, i)).toArray();
    }
    return answer;

  }

  public static void main(String[] args) {

    System.out.println("Hello, World!");
  }
}
